lib/room.rb

```diff
@@ -0,0 +1,37 @@
+module HotelSystem
+  class Room
+    attr_reader :number
+    attr_accessor :reservations,  :block_id, :price
+
+    def initialize(number:, block_id: nil)
+      @number = number
+      @price = 200.00
+      @reservations = []
+      @block_id = block_id
+    end
+
+    def date_available?(possible_start_date, possible_end_date)
+      @reservations.each do |reservation|
+        if possible_start_date == reservation.start_date
+          return false
+        elsif reservation.start_date < possible_start_date && possible_start_date < reservation.end_date
+          return false
```

Comment:
You can simplify all of this complex conditional logic into one line! See the [instructor implementation](https://github.com/droberts-ada/hotel/blob/dpr/solution/lib/date_range.rb#L17).

Each of these would make an excellent test case though.